x/evm/types/genesis_test.go

```diff
@@ -3,18 +3,36 @@ package types
 import (
 	"testing"
 
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
 	ethcmn "github.com/ethereum/go-ethereum/common"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
-	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 
 	"github.com/cosmos/ethermint/crypto/ethsecp256k1"
 )
 
-var address = ethcmn.BytesToAddress([]byte{1, 2, 3, 4, 5})
+type GenesisTestSuite struct {
+	suite.Suite
 
-func TestValidateGenesisAccount(t *testing.T) {
+	address ethcmn.Address
+	hash    ethcmn.Hash
+	code    string
+}
+
+func (suite *GenesisTestSuite) SetupTest() {
+	priv, err := ethsecp256k1.GenerateKey()
+	suite.Require().NoError(err)
+
+	suite.address = ethcmn.BytesToAddress(priv.PubKey().Address().Bytes())
+	suite.hash = ethcmn.BytesToHash([]byte("hash"))
+	suite.code = ethcmn.Bytes2Hex([]byte{1, 2, 3})
+}
+
+func TestGenesisTestSuite(t *testing.T) {
+	suite.Run(t, new(GenesisTestSuite))
+}
+
+func (suite *GenesisTestSuite) TestValidateGenesisAccount() {
 	testCases := []struct {
```

Comment:
need to run `suite.SetupTest()` to instantiate the suite code and hash?

Reply:
not necessary, that is just to reset the values between each run